src/screen.rs

```diff
@@ -1114,22 +1118,53 @@ impl Screen {
             Cancel => {
                 self.error_file = None;
                 self.set_search(None);
+                self.clear_repeat_count();
                 self.error = None;
                 self.refresh();
                 return DisplayAction::ClearOverlay;
             }
             PreviousFile => return DisplayAction::PreviousFile,
             NextFile => return DisplayAction::NextFile,
-            ScrollUpLines(n) => self.scroll_up(n),
-            ScrollDownLines(n) => self.scroll_down(n),
-            ScrollUpScreenFraction(n) => self.scroll_up_screen_fraction(n),
-            ScrollDownScreenFraction(n) => self.scroll_down_screen_fraction(n),
-            ScrollToTop => self.scroll_to(0),
-            ScrollToBottom => self.following_end = true,
-            ScrollLeftColumns(n) => self.scroll_left(n),
-            ScrollRightColumns(n) => self.scroll_right(n),
-            ScrollLeftScreenFraction(n) => self.scroll_left_screen_fraction(n),
-            ScrollRightScreenFraction(n) => self.scroll_right_screen_fraction(n),
+            ScrollUpLines(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_up(n)
+            }
+            ScrollDownLines(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_down(n)
+            }
+            ScrollUpScreenFraction(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_up_screen_fraction(n)
+            }
+            ScrollDownScreenFraction(n) => {
+                let n = self.apply_repeat_count(n);
+                self.scroll_down_screen_fraction(n)
+            }
+            ScrollToTop => self.scroll_to(self.repeat_count.unwrap_or(0)),
```

Comment:
One notable caveat here is that `scroll_to` aims to have the target line in the center of the screen, which doesn't match `less` which puts it at the top of the screen.  We could potentially have two different kinds of absolute scroll: `Top` and `Center`, and use `Top` here.

Reply:
`less` or `vim` does not seem to put the line at the top of the screen. vim seems to center it too.